### PR TITLE
User coordinates

### PR DESCRIPTION
**Pull request by ethanarsht**
## Describe your changes
Bunch of stuff. Modification to distance calculation function in GIS module.
Addition of latitude and longitude to onboarding model, and migration to add those columns to database.
Addition of distance calculation for inference pipeline.
Addition of filters for the query to get the top 2 events.

## Checklist before merging

- [x ] My code produces no errors when run
- [ x] My code is linted and properly formatted

### CODE REVIEW

File: backend/shoulder/s2s/views.py [hunk 1]
```diff
@@ -1611,12 +1616,37 @@ def trigger_suggestions(self, request):
         if update_response.status_code != 200:
```
Comment (chanteriam) [inline backend/shoulder/s2s/views.py:1616]: Should the predictions be updated each time a user suggestion get request is made? I think we came to the conclusion that updates should only happen on a weekly-ish basis, and instead suggestions should be taken directly from the EventSuggestions table for this endpoint

Comment (chanteriam) [reply backend/shoulder/s2s/views.py:1616]: I think instead line 1609/1614 should be `update_response = SuggestionResults.objects.get(user_id=user_id)`

**Review by chanteriam — CHANGES_REQUESTED** [review]

File: backend/shoulder/s2s/views.py (hunk 1 above)
Comment (chanteriam) [reply backend/shoulder/s2s/views.py:1616]: Also, as part of the response for the `trigger_suggestions()` function, can you also include how far the event is from the user? You should be able to do this by modifying lines 1663-1675 and adding something like `event_suggestion['distance_from_user'] = next(key for key, value in self.distance_calc(event_id, user_id).items() if value)` which gets the first key where the distance within is true; so if the returned dictionary is 
```python
{'dist_within_1mi': False,
 'dist_within_5mi': True,
 'dist_within_10mi': True,
 'dist_within_15mi': True,
 'dist_within_20mi': True,
 'dist_within_30mi': True,
 'dist_within_40mi': True,
 'dist_within_50mi': True}
```

then the code would return 'dist_within_5mi'

Comment (ethanarsht) [reply backend/shoulder/s2s/views.py:1616]: That's right! will implement

Comment (ethanarsht) [reply backend/shoulder/s2s/views.py:1616]: Re distance, also implementing!

**Review by chanteriam — APPROVED** [review]

**Review by chanteriam — APPROVED** [review]

File: backend/shoulder/s2s/views.py [hunk 2]
```diff
@@ -1604,19 +1609,38 @@ def trigger_suggestions(self, request):
...
         two_weeks_from_now = current_date + timedelta(days=14)
-        top_events = SuggestionResults.objects.filter(user_id = user_id,
-                                                    event_id__datetime__date__range=(current_date, two_weeks_from_now)) \
-                                                    .order_by('-probability_of_attendance') \
-                                                    .values('event_id', 'probability_of_attendance', 'user_id')[:2]
+        
+        already_rsvp_subquery = UserEvents.objects.filter(
+            user_id_id=OuterRef('user_id'), 
+            event_id_id=OuterRef('event_id')
+        )
```
Comment (chanteriam) [inline backend/shoulder/s2s/views.py:1623]: Add a filter to only look at events that are within 50 miles of the user

**Review by chanteriam — CHANGES_REQUESTED** [review]

**Pull request merged**